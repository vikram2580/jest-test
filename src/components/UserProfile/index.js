import React from "react";

const UserProfile = ({ user }) => {
  return (
    <div>
      {user ? (
        <div>
          <h1>{user.name}</h1>
          <p>{user.email}</p>
        </div>
      ) : (
        <p>No user information available.</p>
      )}
    </div>
  );
};

export default UserProfile;
