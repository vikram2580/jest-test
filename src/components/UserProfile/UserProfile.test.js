import React from "react";
import renderer from 'react-test-renderer';
import UserProfile from ".";

it('renders corrrectly when there is a user',()=>{
    const user ={name: 'Vikram', email: 'vikram@exapmple.com'};
    const tree = renderer.create(<UserProfile user={user} />).toJSON();
    expect(tree).toMatchSnapshot();
})
it('renders corrrectly when there is a user',()=>{
    const tree = renderer.create(<UserProfile/>).toJSON();
    expect(tree).toMatchSnapshot();
})