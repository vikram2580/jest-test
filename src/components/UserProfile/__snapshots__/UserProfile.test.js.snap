// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders corrrectly when there is a user 1`] = `
<div>
  <div>
    <h1>
      Vikram
    </h1>
    <p>
      vikram@exapmple.com
    </p>
  </div>
</div>
`;

exports[`renders corrrectly when there is a user 2`] = `
<div>
  <p>
    No user information available.
  </p>
</div>
`;
